samples/25.message-reaction/src/main/java/com/microsoft/bot/sample/messagereaction/MessageReactionBot.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.messagereaction;
+
+import com.microsoft.bot.builder.ActivityHandler;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.MessageReaction;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+
+public class MessageReactionBot extends ActivityHandler {
+	private static final ActivityLog log;
+	
+	public MessageReactionBot(ActivityLog withLog) {
+            log = withLog;
+	}
+
+	@Override
+	protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+            return sendMessageAndLogActivityId(turnContext, String.format("echo: %s", turnContext.getActivity().getText()));
+	}
+	
+	@Override
+	protected CompletableFuture<Void> onReactionsAdded(List<MessageReaction> messageReactions, TurnContext turnContext) {
+            for (MessageReaction reaction : messageReactions) {
+                // The ReplyToId property of the inbound MessageReaction Activity will correspond to a Message Activity which
+                // had previously been sent from this bot.
+                return log.find(turnContext.getActivity().getReplyToId())
+                        .thenCompose(resultActivity -> {
+                        if (resultActivity == null)
+                        {
+                            // If we had sent the message from the error handler we wouldn't have recorded the Activity Id and so we
+                            // shouldn't expect to see it in the log.
+                            sendMessageAndLogActivityId(turnContext, String.format("Activity %s not found in the log.", turnContext.getActivity().getReplyToId()));
+                        }
+                        
+                        sendMessageAndLogActivityId(turnContext, String.format("You added '%s' regarding '%s'", reaction.getType(), resultActivity.getText()));
+                    }
+                );
+            }
+            
+            return CompletableFuture.completedFuture(null);
+	}
+        
```

Comment:
Are you missing [OnReactionsRemovedAsync](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/25.message-reaction/Bots/MessageReactionBot.cs#L44) method?